### PR TITLE
conformance tests: allow scenarios to add multiple NDs

### PR DESCRIPTION
**Pull request by kdomanski**
**What this PR does / why we need it**:
To run conformance scenarios with multiple availability zones.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: api/cmd/conformance-tests/runner.go [hunk 1]
```diff
@@ -53,7 +54,8 @@ var (
 type testScenario interface {
 	Name() string
 	Cluster(secrets secrets) *kubermaticv1.Cluster
-	Nodes(num int, secrets secrets) *kubermaticapiv1.NodeDeployment
+	Nodes(num int, dc provider.DatacenterSpec, secrets secrets) []kubermaticapiv1.NodeDeployment
```
Comment (alvaroaleman) [inline api/cmd/conformance-tests/runner.go]: Node introduced by you, but could we rename this to `NodeDeployments`?

File: api/cmd/conformance-tests/runner.go [hunk 2]
```diff
@@ -537,65 +543,84 @@ func (r *testRunner) setupNodes(parentLog *logrus.Entry, scenarioName string, cl
...
-		return fmt.Errorf("failed to get MachineDeployment from NodeDeployment: %v", err)
-	}
 
-	log = log.WithField("machineDeployment", machineDeployment.Name)
-	if err := retryNAttempts(defaultAPIRetries, func(attempt int) error {
-		if err := client.Create(ctx, machineDeployment); err != nil {
-			if kerrors.IsAlreadyExists(err) {
-				return nil
+		// we try to subract the non-owned machines sort-of uniformly from all node deployments
+		for machinesWithoutOwner > 0 {
```
Comment (alvaroaleman) [inline api/cmd/conformance-tests/runner.go]: I think we can remove the whole code to handle machines without owner. It was needed for upgrade tests from a version of the conformance-tester that didn't use NodeDeployments yet. The one in 2.10 has it already, thought.

File: api/cmd/conformance-tests/runner.go [hunk 3]
```diff
@@ -822,13 +847,18 @@ func (r *testRunner) getGinkgoRuns(
...
 	cluster *kubermaticv1.Cluster,
-	nd *kubermaticapiv1.NodeDeployment,
+	nd []kubermaticapiv1.NodeDeployment,
 	dc provider.DatacenterMeta,
 ) ([]*ginkgoRun, error) {
 	kubeconfigFilename = path.Clean(kubeconfigFilename)
 	repoRoot := path.Clean(r.repoRoot)
 	MajorMinor := fmt.Sprintf("%d.%d", cluster.Spec.Version.Major(), cluster.Spec.Version.Minor())
 
+	nodesTotal := int32(0)
```
Comment (alvaroaleman) [inline api/cmd/conformance-tests/runner.go]: Can we rename this to `nodNumberTotal`?

File: api/cmd/conformance-tests/runner.go [hunk 4]
```diff
@@ -876,19 +906,31 @@ func (r *testRunner) getGinkgoRuns(
...
 		}
 
 		args = append(args, "--provider=local")
 
-		if nd.Spec.Template.OperatingSystem.Ubuntu != nil {
+		// verify that all operating system specs are identical
+		var osSpec *kubermaticapiv1.OperatingSystemSpec
+		for i, ndi := range nd {
+			if osSpec == nil {
+				osSpec = &ndi.Spec.Template.OperatingSystem
```
Comment (alvaroaleman) [inline api/cmd/conformance-tests/runner.go:892]: Why is this needed? Shouldn't it be enough to verify that the operating system is identical?

Comment (kdomanski) [reply api/cmd/conformance-tests/runner.go:892]: If I compare the whole OS spec then there's no need for comparing every OS's spec to nil in order to determine which OS is actually configured.

**Comment by kdomanski** [comment]
/hold
Let's wait for the new release branch split before merging this.
